library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -484,34 +484,40 @@ public FSFilterStreamContext load(@Nonnull String filename)
         //of this operator.
         for(String seenFileName: endOffsets.keySet()) {
           try {
-            Integer part = openPart.get(seenFileName).getValue() + 1;
+            Integer fileOpenPart = this.openPart.get(seenFileName).getValue();
+            int nextPart = fileOpenPart + 1;
```

Comment:
We can't delete old tmp files here because when the operator is restored to  a previous state after recovery those tmp files will be needed. It is mentioned in the comment. The old tmp files are deleted in committed call back.